Build fails on mips architectures

Another issue found as LXD hits the Debian buildds. The `mips64el` and `mipsel` builds encounter a compiler error:

```
# github.com/lxc/lxd/lxd/instance/drivers
src/github.com/lxc/lxd/lxd/instance/drivers/driver_lxc.go:7129:89: cannot use stat.Dev (variable of type uint32) as type uint64 in argument to unix.Major
src/github.com/lxc/lxd/lxd/instance/drivers/driver_lxc.go:7129:111: cannot use stat.Dev (variable of type uint32) as type uint64 in argument to unix.Minor
```

Apparently on mips architectures the size of `stat.Dev` is only 32 bits ([ref](https://go.googlesource.com/go/+/refs/heads/master/src/syscall/ztypes_linux_mips64le.go#100)) unlike most other platforms where it is 64 bits ([ref](https://go.googlesource.com/go/+/refs/heads/master/src/syscall/ztypes_linux_amd64.go#102)). This then causes a compiler error when a uint32 is passed into the `unix.Major()` and `unix.Minor()` functions in the `x/sys` package, which only seems to have a version that accepts a uint64 ([ref](https://go.dev/src/cmd/vendor/golang.org/x/sys/unix/dev_linux.go#L21)).

There is a comment at the top of the `dev_linux.go` file describing how the encoding of `dev_t` in glibc is backwards-compatible with 32 and 16 bit representations, and it's safe to cast a uint32 up to a uint64, so I think fixing this might be as simple as adding an explicit cast to the code in `driver_lxc.go`. (Backing this up is that every other call to `unix.Major()` and `unix.Minor()` in LXD's code is already doing this.) However, I don't want to just blindly patch this in case there was some specific reason it hasn't been done here. Plus, getting it fixed upstream is one less patch to maintain in Debian's packaging. :wink: 

_(Update 9/14 -- While reproducing this locally with qemu, I had to add `mipsle` and `mips64le` to the architecture aliases list in `shared/osarch/architectures.go`. I don't know if that was due to a quirk of how qemu presents the virtualized mips environment or not, but it probably wouldn't hurt to add those two aliases as well.)_

Buildd logs:
https://buildd.debian.org/status/fetch.php?pkg=lxd&arch=mips64el&ver=5.0.0-1&stamp=1662903156&raw=0
https://buildd.debian.org/status/fetch.php?pkg=lxd&arch=mipsel&ver=5.0.0-1&stamp=1662903038&raw=0

--- a/lxd/instance/drivers/driver_lxc.go
+++ b/lxd/instance/drivers/driver_lxc.go
@@ -7103,7 +7103,7 @@ func (d *lxc) getFSStats() (*metrics.MetricSet, error) {
 					return nil, fmt.Errorf("Failed to stat %s: %w", dev["source"], err)
 				}
 
-				backingFilePath := fmt.Sprintf("/sys/dev/block/%d:%d/loop/backing_file", unix.Major(stat.Dev), unix.Minor(stat.Dev))
+				backingFilePath := fmt.Sprintf("/sys/dev/block/%d:%d/loop/backing_file", unix.Major(uint64(stat.Dev)), unix.Minor(uint64(stat.Dev)))
 
 				if shared.PathExists(backingFilePath) {
 					// Read backing file

